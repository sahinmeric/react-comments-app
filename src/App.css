.App {
  text-align: center;
}

body {
  margin: 0;
  padding: 0;
}

ul {
  list-style-type: none;
  padding: 0;
  margin: 0;
}
