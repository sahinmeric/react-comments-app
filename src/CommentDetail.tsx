import { useNavigate } from "react-router-dom";

function CommentDetail() {
    const navigate = useNavigate()

    return (
        <div>
            <p>This is comment detail page</p>
            <button onClick={() => navigate(-1)}>Back</button>
        </div>
    )
}

export default CommentDetail