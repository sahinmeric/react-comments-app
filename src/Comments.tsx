import React from "react";
import axios from "axios";
import { useQuery } from "@tanstack/react-query";

import NotFound from "./NotFound";
import ShowProgress from "./ShowProgress";
import CommentCard from "./CommentCard"
import { Typography } from "@mui/material";


export type CommentType = {
    commentId: number;
    id: number;
    name: string;
    email: string;
    body: string;
};

export function useComments() {
    return useQuery(["comments"], async (): Promise<Array<CommentType>> => {
        const { data } = await axios.get(
            "http://jsonplaceholder.typicode.com/posts/1/comments"
        );
        return data;
    });
}

function Comments() {
    const { data: comment, error, isFetching, isLoading } = useComments();
    if (isLoading || isFetching) {
        return <ShowProgress />
    }
    if (error || !comment) {
        return <NotFound />
    }
    return (
        <>
            <ul>
                <Typography
                    variant="h3"
                    style={{
                        padding: 10,
                        margin: "auto",
                        width: "50%",
                        textAlign: "center",
                        color:"#192F52"
                    }}>
                    Comments
                </Typography>
                {comment?.map((comment) => (
                    <li key={comment.id}>
                        <CommentCard
                            commentId={comment.commentId}
                            id={comment.id}
                            name={comment.name}
                            email={comment.email}
                            body={comment.body}
                        />
                    </li>
                ))}
            </ul>
        </>
    );
}

export default Comments;
