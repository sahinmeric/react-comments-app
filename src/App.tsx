import React from 'react';
import './App.css';
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
import { Route, Routes } from "react-router-dom";
//components
import Comments from "./Comments";
import CommentDetail from './CommentDetail';
import NotFound from './NotFound';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            cacheTime: 1000 * 60 * 60 * 24,
            refetchOnWindowFocus: false,
            refetchOnMount: false,
        },
    },
});

function App() {

    return (
        <QueryClientProvider client={queryClient}>
            <Routes>
                <Route path='/' element={<Comments />} />
                <Route path='/:commentId' element={<CommentDetail />} />
                <Route path='*' element={<NotFound />} />
            </Routes>
            <ReactQueryDevtools initialIsOpen />
        </QueryClientProvider>
    );
}

export default App;
