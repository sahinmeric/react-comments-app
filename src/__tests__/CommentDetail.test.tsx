/**
 * @jest-environment jsdom
 */
import { describe, expect, test } from '@jest/globals';
import { queryByTestId, render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import CommentDetail from '../CommentDetail';
import Comments from '../Comments';
import CommentCard from '../CommentCard';
import App from "../App"
import { BrowserRouter } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import renderer from 'react-test-renderer';


const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            cacheTime: 1000 * 60 * 60 * 24,
            refetchOnWindowFocus: false,
            refetchOnMount: false,
        },
    },
});

describe('CommentDetail test suit', () => {
    beforeAll(() => {
        /* Runs before all tests */
    })
    afterAll(() => {
        /* Runs after all tests */
    })
    beforeEach(() => {
        /* Runs before each test */
    })
    afterEach(() => {
        /* Runs after each test */
    })

    test('if it renders without crash', () => {
        const component = renderer.create(<BrowserRouter><QueryClientProvider client={queryClient}><CommentDetail /></QueryClientProvider></BrowserRouter>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // test('should render a button with the class of primary', () => {
    //     render(<CommentDetail />)
    // })
    // test('should render a button with the class of primary', () => {
    //     render(<CommentCard commentId={1}
    //         id={2}
    //         name={"asdasda"}
    //         email={"email"}
    //         body={"this is comentario"}/>)
    // })

});

