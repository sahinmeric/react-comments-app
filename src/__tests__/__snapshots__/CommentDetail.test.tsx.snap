// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentDetail test suit if it renders without crash 1`] = `
<div
  className="MuiBox-root css-0"
  style={
    {
      "left": "50%",
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={{}}
      />
    </svg>
  </span>
</div>
`;
